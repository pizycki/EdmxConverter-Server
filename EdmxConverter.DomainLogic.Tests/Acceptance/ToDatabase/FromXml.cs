using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using EdmxConverter.DomainLogic.Service;
using EdmxConverter.DomainLogic.Tests.Properties;
using Xunit;
using static LanguageExt.Prelude;

namespace EdmxConverter.DomainLogic.Tests.Acceptance.ToDatabase
{
    public class FromXml
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [UseApprovalSubdirectory("Approved")]
        public void convert_from_plain_xml() =>
            Some(Resources.SampleXmlEdmx)
                .Map(sample => new XmlEdmx(sample))
                .Bind(ConvertToDatabase.FromXml)
                .Match(edmx => edmx.ToString(), () => None.ToString())
                .Verify();
    }
}