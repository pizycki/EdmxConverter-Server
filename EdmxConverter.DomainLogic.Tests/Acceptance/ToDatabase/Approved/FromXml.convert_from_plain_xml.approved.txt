0x1F8B0800000000000400CD57DB6EDC36107D2FD07F20F8D4028E683B2FAD212570D6DEC068D60E2227EF5C69764D942215923256DF96877E527FA143DD2FDEDD38098262818548CD1CCE1C9E1952FF7EF9277CBDCB247904638556113D0B4E290195E854A86D440BB779F1077DFDEAD75FC2EB34DB914FADDD4B6F879ECA46F4C1B9FC82319B3C40C66D9089C468AB372E4874C678AAD9F9E9E99FECEC8CA5DCF135B7E0A1286212127E2894131954031C2EB44A207705972B9D82B4CD3CBE892B7472CB33B0394F20A231CF7209570D66D00CD708E160E728B99482637431C80D255C29EDB8C3D82F3E5A889DD16A1BE738C1E57D9903DA6DB8B4D0E474D19B7F6D7AA7E70C521C778E2D545258A7B367029EBDF47C79C0A9FBF3594714DAF1884C5E23E3AEF459576C4674A93525D3752E16D2789B633C07159C001B20CC09191B9F7482415DF9DF095914D21506220585335C9E90F7C55A8AE42F28EFF5DFA0225548390C1703C677A3099C7A6F740EC6951F60D324719352C2C67E6CEAD8B90D7CEA1CDF16029F6F716DBE96D089811D74F7FF2D002A0A0B869215DFBD03B5750F11C5474A966207693BD3A07E5402EB0B9D9C29268B84ACDF9DF99E79CAB950609A0066929F6D226ABDD947DB2C37CEA8C68DC1F552B094F431D4E513540A792ACE2EA2BE50595DA96D45B33D251DAE789E236583126F66485CD7F7E245FC7CB16735064BEC139AEFA2ED5672DAF02D4CDEE2D218E95218EB5A2153B248B399D994FF3DDCB6AB0D289EEAB967BC35F6CFA35DDE577F7E6F26703D974B4C2F03E5AA4CA18BA8AEF99957D566B9E4E6893A596859646A5FAD1DF2AECB64E85FCFCC114236097B4A139BF134A9F829E787243B35E956EFA43B9168D8C8E5F8D134D34F6D420952F328D24A3BA5759005DE20883FCB8514986F6FB0E24A6CC0BABA29523C11CE2727DAFFE77461D6A6F2F811F3D39B7AA1C4E702904D8C6623C07C5783578FDC240FDCFC96F1DDEF43A4EF6DE2F35E73B445EFEBD0B5CA229AAE71705F875777F46FECDD73C1876C78630BAFC08A6D0FE1EF6F0A12AFA41EB4B5B9511BDD128CE90C236A4D26FCAFC0717F67BC34B8813C71F83A016BAB83F64DE9E0D2185E46F43A5B437AA3EE0A9717EED25AC8D6B21CE61CB2C3315487D438EEF02EF723FB23D2F8C4A5C034E04EBD29844C47B12FE73ADC07E385F21670BEAA54BC7020E4B61CA1DD6AF595600D95579083F205720F78AA20A0BD53317F846F8D11EF1BEF60CB93B2ED6387818E6FCC781BC22BC1B78667B6C1E8FDFD170AF3DF15AFFE03F7E85A3FD40C0000